client/python/lib/test/test_grpc_responses.py

```diff
@@ -0,0 +1,155 @@
+        model_raw_responses.append(create_model_status_response(version, status['error_code'], status['error_message'],
+         status['state']))
+    raw_response = merge_model_status_responses(model_raw_responses)
+
+    response = GrpcModelStatusResponse(raw_response)
+    response_dict = response.to_dict()
+    assert isinstance(response_dict, dict)
+    assert len(response_dict) == len(model_raw_status_response_dict)
+    for version, status in model_raw_status_response_dict.items():
+        assert version in response_dict
+        assert isinstance(response_dict[version], dict)
+        assert response_dict[version]['error_code'] == status['error_code']
+        assert response_dict[version]['error_message'] == status['error_message']
+        assert response_dict[version]['state'] == ModelVersionStatus.State.Name(status['state'])
+
+@pytest.mark.parametrize("model_raw_metadata_response_dict" , MODEL_METADATA_RESPONSE_VALID)
+def test_ModelMetadataResponse_to_dict_valid(model_raw_metadata_response_dict):
+    raw_response = create_model_metadata_response(model_raw_metadata_response_dict)
+
+    response = GrpcModelMetadataResponse(raw_response)
+    response_dict = response.to_dict()
+    assert isinstance(response_dict, dict)
+    assert len(response_dict) == 1
+
+    version = model_raw_metadata_response_dict['version']
+    assert version in response_dict
+    assert 'inputs' in response_dict[version] and 'outputs' in response_dict[version]
+
+    inputs = response_dict[version]['inputs']
+    assert len(inputs) == len(model_raw_metadata_response_dict['inputs'])
+    for input in inputs:
+        assert input in model_raw_metadata_response_dict['inputs']
+        assert isinstance(inputs[input], dict)
+        assert 'shape' in inputs[input] and 'dtype' in inputs[input]
+        assert inputs[input]['shape'] == model_raw_metadata_response_dict['inputs'][input]['shape']
+        assert inputs[input]['dtype'] == DataType.Name(model_raw_metadata_response_dict['inputs'][input]['dtype'])
+
+
+    outputs = response_dict[version]['outputs']
+    assert len(outputs) == len(model_raw_metadata_response_dict['outputs'])
+    for output in outputs:
+        assert output in model_raw_metadata_response_dict['outputs']
+        assert isinstance(outputs[output], dict)
+        assert 'shape' in outputs[output] and 'dtype' in outputs[output]
+        assert outputs[output]['shape'] == model_raw_metadata_response_dict['outputs'][output]['shape']
+        assert outputs[output]['dtype'] == DataType.Name(model_raw_metadata_response_dict['outputs'][output]['dtype'])
+
+def create_model_metadata_response(model_raw_metadata_response_dict):
+    raw_response = GetModelMetadataResponse()
+
+    model_spec = ModelSpec()
+    model_spec.name = model_raw_metadata_response_dict['name']
+    model_spec.version.value = model_raw_metadata_response_dict['version']
+
+    raw_response.model_spec.CopyFrom(model_spec)
+
+    metadata = SignatureDef()
```

Comment:
You use `metadata` name below and reassign different type to it. 
Perhaps we could go with just `signature_def` here?